CONTRIBUTING.md

```diff
@@ -4,45 +4,87 @@ Above all, thank you for your interest in both the Synthstrom Deluge and being a
+
+
+# Code review and testing
+
+Pull requests that are no longer marked as Draft and are open for merging should be ready to test and any help is appreciated. 
+
+* This includes both building and testing the firmware as well as reading and reviewing the source code.
+* Please use the review and discussion functionality provided within GitHub.
+* Code reviews are intended to increase quality and should not go into nitpicking, please remember the [Code of Conduct](CODE_OF_CONDUCT.md).
+* Final decisions about merging of Pull requests is up to the code owners, see below and in [Governance](GOVERNANCE.md).
+
+
+# Repository contributions
+
+Everybody is invited to contribute to this repository as further outlined below:
 
-## Pull Requests (PR)
+* All contributions are handled with the [Pull request](https://github.com/SynthstromAudible/DelugeFirmware/pulls) mechanic in GitHub. If you start working on something please submit a Draft PR so everyone is informed (see below).
+* Contributions can be anything that improves the project while fulfilling the requirements outlined below.
+* As an inspiration here is a small and incomplete list of improvement areas: documentation, commenting complex code mechanics, refactoring (make code easier to read/improve clarity/make execution more efficient/improve maintainability/ease modification/improve structure), feature improvements, new features, easier accessibility, toolchain improvements, unit tests and many, many more.
+* For information on the decision making process for accepting Pull requests please see below and also see the [Governance](GOVERNANCE.md) documentation.
 
-* Work-in-progress should be submitted as either a self-assigned issue, a *Draft PR*, or both (ideally, linked to each other). Draft PRs will be considered the *official* way work-in-progress is tracked.
 
-* Try to name your branch appropriately and keep the scope of changes clear (eg: if working on a specific issue, try not to incorporate other unrelated changes or fixes into the same PR)
+## Requirements
 
-* As the *community* branch gets updated, please keep your PR's branch updated so it can be easily merged
+The following requirements must be fulfilled for a Pull request to be mergable to the community branch:
 
-* As formatting standards are developed, please ensure you are formatting your code accordingly. clang formatting has been included in the GitHub Actions to help with this.
+### General
 
-### New features
+* The contribution must be meaningful to the project. There must be a clear and articulate link between the change and an improvement for the developers, users or overall community.
+* The Pull request must have a clear scope outlined in the description, please don't commit changes unrelated to the scope.
+* Appropriate branch name, if possible following standard conventions like git flow (e.g. feature/shiny_new_feature_name).
```

Comment:
I've tended to use `<username>/<feature_area>` myself, but am happy to change it up. I know most will be submitting from forks so `feature/feature_name_or_domain` is a good approach.